### PR TITLE
Add possibility of changing the Menu and setting the home page of My Account

### PR DESCRIPTION
**Pull request by klzns**
#### What is the purpose of this pull request?

Make it possible for an app to:
1. Change the Menu implementation.
2. Change the default page of My Account.

#### What problem is this solving?

1. Menu customization.
2. Default page opened when the user goes to `/account/`.

#### How should this be manually tested?

Follow the instructions provided by the README.

#### Screenshots or example usage

Custom Menu:
image

#### Types of changes

- [x] New feature (a non-breaking change which adds functionality)
- [x] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,88 +2,104 @@
...
 
-Add the dependency in your `manifest.json`
+This app provides a few extension points in order to allow stores to customize their customer's experience as needed. You can increment this app with more pages with custom navigation using `react-router-dom`.
 
-```json
-"dependencies": {
-  "vtex.my-account": "x.x"
-}
-```
+To do so you gonna need two components, `ExtensionLinks` and `ExtensionRouter`.
```
Comment (kevinch) [inline README.md]: Nitpick: "To do so you will need two components". `gonna` is really informal, especially written.

File: README.md [hunk 2]
```diff
@@ -2,88 +2,104 @@
 
 My account is a canonical app that any VTEX store can use. This app is responsible for managing the customer's personal data such as his addresses and credit cards.
 
-## Setup
+## Extension points
 
-Add the dependency in your `manifest.json`
+This app provides a few extension points in order to allow stores to customize their customer's experience as needed. You can increment this app with more pages with custom navigation using `react-router-dom`.
```
Comment (salesfelipe) [inline README.md]:
```suggestion
This app provides a few extension points in order to allow stores to customize their customer's experience as needed. You can increment this app with more pages with your own custom navigation using `react-router-dom`.
```

File: README.md (hunk 1 above)
Comment (salesfelipe) [inline README.md]:
```suggestion
To do so you gonna need two components: `ExtensionLinks`, `ExtensionRouter`. 
```

File: react/components/Menu/index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  render() {
+    const { hasExtension } = this.state
+
+    return (
+      <aside className="vtex-account__menu pv9 pv0-m-2 ph9 ph7-m ph8-l w-20-m w-100">
+        <UserInfo />
+        <ExtensionPoint id="menu" hasExtension={this.handleExtension} />
+        {hasExtension !== null && hasExtension === false && <MenuLinksList />}
```
Comment (kevinch) [inline react/components/Menu/index.js]: Is `hasExtension !== null` useful considering the second verification `hasExtension === false`?

Comment (klzns) [reply react/components/Menu/index.js]: Nope, there are three different states here:

1. `hasExtension` being `null`
2. `hasExtension` being `false`
3. `hasExtension` being `true`

**Pull request merged**